src/averageAges.js

```diff
@@ -55,7 +67,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => people.find(
+    mother => mother.name === person.mother)
+    && (onlyWithSon ? person.sex === 'm' : true));
+
+  const ageDiff = children.map(child => {
+    const mother = people.find(person => child.mother === person.name);
+
+    return child.born - mother.born;
+  });
+
+  const avgAgeDiff = ageDiff.reduce(
```

Comment:
It's better to not shortcut the word in variable name, so it can be easy to read them